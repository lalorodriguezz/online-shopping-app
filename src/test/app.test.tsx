import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import RecipeCard from "../components/ui/card";
import "@testing-library/jest-dom";

describe("Testing RecipeCard.tsx", () => {
  it("Debe renderizar correctamente el componente", () => {
    const { baseElement } = render(<RecipeCard />);
    expect(baseElement).toMatchSnapshot(); // Aseguramos que el componente no ha cambiado
  });

  it("Debe mostrar el título 'Online Shopping'", () => {
    render(<RecipeCard />);
    expect(screen.getByText(/Online Shopping/i)).toBeInTheDocument();
  });

  it("Debe mostrar la descripción correcta", () => {
    render(<RecipeCard />);
    expect(
      screen.getByText(/You can shopping anytime, anywhere that you want./i)
    ).toBeInTheDocument();
  });

  it("Debe renderizar un botón con el texto 'Shopping Now'", () => {
    render(<RecipeCard />);
    const button = screen.getByRole("button", { name: /Shopping Now/i });
    expect(button).toBeInTheDocument();
  });

  it("Debe disparar el evento al hacer clic en el botón", () => {
    const handleClick = vi.fn(); // Mock de la función de clic
    render(
      <button onClick={handleClick} className="btn-test">
        Shopping Now
      </button>
    );

    const button = screen.getByRole("button", { name: /Shopping Now/i });
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
