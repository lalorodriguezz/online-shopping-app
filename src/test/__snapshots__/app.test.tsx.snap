// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing RecipeCard.tsx > Debe renderizar correctamente el componente 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center min-h-screen bg-gradient-to-b from-blue-50 via-white to-blue-100 px-4"
    >
      <div
        class="bg-white p-8 lg:p-12 rounded-3xl shadow-xl max-w-md w-full text-center transform hover:scale-105 transition-transform duration-300"
      >
        <div
          class="w-44 h-44 lg:w-52 lg:h-52 mx-auto overflow-hidden rounded-2xl"
        >
          <img
            alt="Online Shopping"
            class="w-full h-full object-cover"
            src="/src/components/ui/images/imagen1.png"
          />
        </div>
        <h2
          class="text-3xl lg:text-4xl font-extrabold text-gray-900 mt-6"
        >
          Online Shopping
        </h2>
        <p
          class="text-gray-500 text-base lg:text-lg mt-3"
        >
          You can shopping  anytime, anywhere that you want.
        </p>
        <button
          class="mt-6 w-[90%] bg-green-500 hover:bg-green-900 text-white font-semibold py-3 px-6 rounded-xl shadow-md hover:shadow-lg transition-all duration-300"
        >
          Shopping Now
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Testing de RecipeCard > Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center min-h-screen bg-gradient-to-b from-blue-50 via-white to-blue-100 px-4"
    >
      <div
        class="bg-white p-8 lg:p-12 rounded-3xl shadow-xl max-w-md w-full text-center transform hover:scale-105 transition-transform duration-300"
      >
        <div
          class="w-44 h-44 lg:w-52 lg:h-52 mx-auto overflow-hidden rounded-2xl"
        >
          <img
            alt="Online Shopping"
            class="w-full h-full object-cover"
            src="/src/components/ui/images/imagen1.png"
          />
        </div>
        <h2
          class="text-3xl lg:text-4xl font-extrabold text-gray-900 mt-6"
        >
          Online Shopping
        </h2>
        <p
          class="text-gray-500 text-base lg:text-lg mt-3"
        >
          You can shop anytime, anywhere that you want.
        </p>
        <button
          class="mt-6 w-[90%] bg-green-500 hover:bg-green-900 text-white font-semibold py-3 px-6 rounded-xl shadow-md hover:shadow-lg transition-all duration-300"
        >
          Shop Now
        </button>
      </div>
    </div>
  </div>
</body>
`;
